graphql/schemabuilder/reflect.go

```diff
@@ -240,7 +240,17 @@ func (sb *schemaBuilder) getEnumArgParser(typ reflect.Type) (*argParser, graphql
 
 }
 
+// cachedType is a container for GraphQL datatype and the list of its fields
+type cachedType struct {
+	argType *graphql.InputObject
+	fields  map[string]argField
+}
+
+// typeCache maps Go types to GraphQL datatypes
+var typeCache map[reflect.Type]cachedType
```

Comment:
can we keep this cache on the schemabuilder instead of sharing one globally?

Reply:
updated